plugins/gcp/resoto_plugin_gcp/resources/base.py

```diff
@@ -301,14 +302,22 @@ class GcpResource(BaseResource):
     link: Optional[str] = None
     label_fingerprint: Optional[str] = None
 
+    @property
+    def _get_identifier(self) -> str:
+        return self.api_spec.accessors[-1][:-1] # Poor persons `singularize(), i.e. ["vpnTunnels"] -> "vpnTunnel"`
```

Comment:
This is not a property of a resource - please remove it. Could you make it a property in api_spec?

Reply:
Yes and no. `GcpResource`s should have more than one "api spec". One for collecting, using list or aggregateList actions, one for `get`ting an individual resource, one for deletion and one for updating tags. Currently we just say that the one api spec is the collection use case. In this PR so far I went down the route to modify (guess, really) the generated api spec to comply with the `setLabels` and `get` use case, where ideally we should have those generated with `model_gen.py` as well.
Mixing the multiple use cases into the one api spec just feels wrong, like having the `get` specific identifier as a property on the otherwise `aggregateList` api spec. What do you think?

Reply:
Heres an update that moves the multiple changes into a the ApiSpec class, sticking with collection use case as default, but extending the interface to get the `get` and `setLabels` specific specs from it as well: b4e21bdeba0869dfb3cffa41e79baab560b156eb

How do you feel about this approach?

Reply:
So yes - now it is static information attached to a resource class - that looks much better.
I also like your idea of creating specs on demand - I hope this works for all resource types?

Reply:
Looks like it so far. There's only one or two special snowflakes where I had to define deviating behaviour 👍🏻